core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -45,70 +38,46 @@
 /**
  * Base class for {@link TableScan} implementations.
  */
-abstract class BaseTableScan implements TableScan {
+abstract class BaseTableScan extends BaseScan<TableScan> implements TableScan {
   private static final Logger LOG = LoggerFactory.getLogger(BaseTableScan.class);
 
-  private final TableOperations ops;
-  private final Table table;
-  private final Schema schema;
-  private final TableScanContext context;
-
   protected BaseTableScan(TableOperations ops, Table table, Schema schema) {
     this(ops, table, schema, new TableScanContext());
   }
 
   protected BaseTableScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
-    this.ops = ops;
-    this.table = table;
-    this.schema = schema;
-    this.context = context;
-  }
-
-  protected TableOperations tableOps() {
-    return ops;
-  }
-
-  protected Schema tableSchema() {
-    return schema;
+    super(ops, table, schema, context);
   }
 
   protected Long snapshotId() {
-    return context.snapshotId();
+    return context().snapshotId();
   }
 
   protected boolean colStats() {
-    return context.returnColumnStats();
+    return context().returnColumnStats();
   }
 
   protected boolean shouldIgnoreResiduals() {
-    return context.ignoreResiduals();
+    return context().ignoreResiduals();
   }
 
   protected Collection<String> selectedColumns() {
-    return context.selectedColumns();
+    return context().selectedColumns();
   }
 
   protected ExecutorService planExecutor() {
-    return context.planExecutor();
+    return context().planExecutor();
   }
 
   protected Map<String, String> options() {
-    return context.options();
-  }
-
-  protected TableScanContext context() {
-    return context;
+    return context().options();
   }
 
-  @SuppressWarnings("checkstyle:HiddenField")
-  protected abstract TableScan newRefinedScan(TableOperations ops, Table table, Schema schema,
-                                              TableScanContext context);
-
   protected abstract CloseableIterable<FileScanTask> doPlanFiles();
 
   @Override
   public Table table() {
```

Comment:
Do we still need this implementation here? Can't we consume from the parent directly?

Reply:
yeah. `BaseScan#table()` is a protected method. this is a public method.